### PR TITLE
Pull Request for project RoverSim

### PR DESCRIPTION
**Pull request by brahatesh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,73 @@
 # RoverSim
 Open Projects 2021
```
Comment (theobscuredev) [inline README.md]: Use ## (H2 heading) for the text 'Open Projects 2021'

File: README.md [hunk 2]
```diff
@@ -1,2 +1,73 @@
...
+
+## Abstract
+
+The project aims to introduce the Path-Planning phase to a wireless rover to make it capable to navigate autonomously in an environment with the help of path planning algorithms (Dijkstra and SLAM) parsed by a python script. It aims to reach the destination through the shortest path and shortest time. This was done using ROS navigation stack by performing a simulation of a rover model in Gazebo environment, by running ROS (Melodic Morenia) in Ubuntu 18.04. The robot used in this project is Turtlebot 3.
+
+![gazebo_sim](https://github.com/brahatesh/RoverSim/blob/main/Images%20and%20Videos/Gifs/gazebo_sim.gif "Simulation in Gazebo")
+
+![rviz_sim](https://github.com/brahatesh/RoverSim/blob/main/Images%20and%20Videos/Gifs/rviz_sim.gif "Simulation in RViz")
```
Comment (theobscuredev) [inline README.md]: Center align all the images using the `<p>` paragraph tag and `align` attribute. For eg:
`<p align="center">`
  `LED Matrix`
  `<i>LED Matrix</i>`
`</p>`

Refer this README for the same: https://raw.githubusercontent.com/marsiitr/Rube-Goldberg-Machine/master/README.md

Also resize a few images by assigning the dimensions. And include these 2 images before abstract as mentioned in the guidelines.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,73 @@
 # RoverSim
 Open Projects 2021
+
+## Abstract
+
+The project aims to introduce the Path-Planning phase to a wireless rover to make it capable to navigate autonomously in an environment with the help of path planning algorithms (Dijkstra and SLAM) parsed by a python script. It aims to reach the destination through the shortest path and shortest time. This was done using ROS navigation stack by performing a simulation of a rover model in Gazebo environment, by running ROS (Melodic Morenia) in Ubuntu 18.04. The robot used in this project is Turtlebot 3.
```
Comment (theobscuredev) [inline README.md]: Justify the entire text using the `<p>` paragraph tag and `justify` attribute. Make sure you justify all the paragraphs separately for the entire content in README. For eg:
`<p align="justify">`
  `<h2>Abstract</h2>`
  `<p>A Rube Goldberg Machine is .........................</p>`
`</p>`

Refer this README for the same: https://raw.githubusercontent.com/marsiitr/Rube-Goldberg-Machine/master/README.md

File: README.md [hunk 4]
```diff
@@ -1,2 +1,73 @@
...
+1. [Aman Vishwakarma](https://github.com/aman-vishwakarma2024)
+2. [Ankit Dudi](https://github.com/fanaticANKIT)
+3. [Brahatesh Vasantha](https://github.com/brahatesh)
+4. [Harshini Sendhil](https://github.com/cicadaa3301-harshini)
+5. [Prashant Gupta](https://github.com/chill3010house)
+
+## Mentors
+
+1. [Pradnesh Pramod Chavan](https://github.com/theobscuredev)
+2. Shubham Goyal
```
Comment (theobscuredev) [inline README.md]: Hyperlink Shubham's Github ID: https://github.com/shubham491981

File: README.md [hunk 5]
```diff
@@ -1,2 +1,73 @@
...
+![rviz_sim](https://github.com/brahatesh/RoverSim/blob/main/Images%20and%20Videos/Gifs/rviz_sim.gif "Simulation in RViz")
+
+
+## Motivation
+
+<p>Watching a robot move on its own without any human input is always fascinating. So, our team of 5 freshers decided to take on the challenge to learn how a robot maps the environment, find its position in the map of the environment and move from one position to another on its own.<br><br>The future will be fully automated with the help of robots and for that automated navigation will be extremely important. So making a small robot move automatically is the first step.</p>
+
+## Workflow
+
+![Workflow](https://github.com/brahatesh/RoverSim/blob/main/Images%20and%20Videos/Images/Workflow.jpg)
```
Comment (theobscuredev) [inline README.md]: Add a section named 'Software Aspect' and elaborate the Navigation & SLAM part briefly and hyperlink their README files here (https://github.com/brahatesh/RoverSim#readme-files-for-elaborative-explanation-of-working).

File: README.md [hunk 6]
```diff
@@ -1,2 +1,73 @@
...
+* [SLAM basics](https://www.youtube.com/watch?v=B2qzYCeT9oQ&list=PLpUPoM7Rgzi_7YWn14Va2FODh7LzADBSm&ab_channel=ClausBrenner)
+
+* [Chapter 10 in Probabilistic Robotics by Dieter Fox, Sebastian Thrun, and Wolfram Burgard](https://books.google.co.in/books/about/Probabilistic_Robotics.html?id=2Zn6AQAAQBAJ&source=kp_book_description&redir_esc=y)
+
+* [SLAM for dummies](https://dspace.mit.edu/bitstream/handle/1721.1/119149/16-412j-spring-2005/contents/projects/1aslam_blas_repo.pdf)
+
+## Readme files for elaborative explanation of working
+
+* [Dijkstra_Path_Planning](https://github.com/brahatesh/RoverSim/blob/main/src/Dijkstra_Path_Planning/README.md)
+* [EKF_SLAM](https://github.com/brahatesh/RoverSim/blob/main/src/EKF%20SLAM/README.md)
```
Comment (theobscuredev) [inline README.md]: As mentioned above, include this after workflow.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,73 @@
...
+* While navigating to its destination, the robot tends to go too close to an obstacle and get stuck in a loop that makes it rotate constantly.<br><br>
+**EKF Slam has some limitations as listed below:**
+* For larger inputs, the application crashes because the time complexity of the algorithm is O(n^2)
+* Featured(point) landmarks can only be detected
+* All densities involved in the algorithm are modelled as Gaussian. Hence the model is assumed to have a normal distribution which introduces some noise.
+* The environment has to be a static environment.
+
+## Future Improvements
+* **Implementation of fast slam in ROS**
+* **Adding ArUco marker detection capability for pose estimation**
```
Comment (theobscuredev) [inline README.md]: Remove the bold text.

File: README.md [hunk 8]
```diff
@@ -1,2 +1,73 @@
...
+
```
Comment (theobscuredev) [inline README.md:76]: Include the presentation file from 'Poster and Report' folder here.

File: src/Dijkstra_Path_Planning/README.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+    roslaunch sim simulation.launch
+
+## THE CODE EXPLAINED
+
+### THE ALGORITHM - ([File](https://github.com/brahatesh/RoverSim/blob/main/src/Dijkstra_Path_Planning/src/dijkstra_pp/scripts/path_planning_dijkstra.py))
+
+Dijkstra's algorithm is an algorithm that finds the path between 2 nodes with the minimum cost. Cost in this case can be treated as the distance between 2 nodes. 
+
+![Flowchart](https://raw.githubusercontent.com/brahatesh/RoverSim/main/Images%20and%20Videos/Images/djikstra_flowchart.jpg)
```
Comment (theobscuredev) [inline src/Dijkstra_Path_Planning/README.md]: Again center align the image and resize it.

File: src/Dijkstra_Path_Planning/README.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
+# PATH PLANNING
```
Comment (theobscuredev) [inline src/Dijkstra_Path_Planning/README.md:1]: Justify each paragraph.

**Review by theobscuredev — CHANGES_REQUESTED** [review]
Hey @brahatesh,
Nice work team. Although please do the following suggested minor changes:

1. Update the README as per the comments made below. Precisely, focus on the paragraph tags, image size and Software Aspect.
2. Change the format of the poster to .jpeg instead of .jpg
3. Rectify my name in the Project Report on the 2nd page :). 
4. Remove the presentation from 'Poster and Report' folder and include it in the README.
5. Center align the images and resize them in the remaining README files as well and justify the text in the same.
6. Add a cost structure similar to this: https://github.com/tushdon2/RL-based-Maze-Solver#cost-structure

**Comment by theobscuredev** [comment]
@brahatesh Lemme know once you've made the final changes. Make sure to tag me (@) in this thread.

**Comment by brahatesh** [comment]
@theobscuredev Do we remove the presentation from "Poster and Report" folder or keep it there and hyperlink it in readme?

**Comment by theobscuredev** [comment]
> @theobscuredev Do we remove the presentation from "Poster and Report" folder or keep it there and hyperlink it in readme?

Remove it from that folder. Just hyperlink it in the README.

**Comment by astrodigger** [comment]
@theobscuredev  changes made! Verify them. Thanks!

**Comment by theobscuredev** [comment]
Nice work!
@SanjeevKrishnan I suggest for merge.

**Comment by SanjeevKrishnan** [comment]
@brahatesh @fanaticANKIT @cicadaa3301-harshini Great Effort! Excellent work!
PR looks clean and perfect! Merging the PR

**Pull request merged**